engine/servermaster/scheduler/scheduler.go

```diff
@@ -99,14 +99,15 @@ func (s *Scheduler) checkCostAllows(
 
 func (s *Scheduler) getConstraint(
 	ctx context.Context,
-	resources []resourcemeta.ResourceID,
+	resources []resourcemeta.ResourceKey,
 ) (model.ExecutorID, error) {
 	var (
 		lastResourceID resourcemeta.ResourceID
 		ret            model.ExecutorID
 	)
-	for _, resourceID := range resources {
-		executorID, hasConstraint, err := s.placementConstrainer.GetPlacementConstraint(ctx, resourceID)
+	for _, resource := range resources {
+		resourceID := resource.ID
+		executorID, hasConstraint, err := s.placementConstrainer.GetPlacementConstraint(ctx, resource.JobID, resourceID)
```

Comment:
It is simpler to pass `ResourceKey` directly.

Reply:
Sure. I will unify the usage of `ResourceKey `.